spec/unit/s3_source_spec.rb

```diff
@@ -28,7 +28,7 @@
     @s3_connection = double(Fog::Storage::AWS)
     @s3_source = Chef::Knife::S3Source.new
 
-    @s3_source.url = "s3://#{@bucket_name}/#{@test_file_path}"
+    @s3_source.url = "s3://s3.amazonaws.com/#{@bucket_name}/#{@test_file_path}"
```

Comment:
This is a good idea but you might want to modify it slightly to accommodate regions. The way you have written it here, it will only work in us-east-1. See http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region